core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -1407,18 +1424,21 @@ private void buildMatViewGraph() {
                         }
 
                         final MatViewRefreshState state = matViewGraph.addView(matViewDefinition);
-                        final boolean isMatViewStateExists = TableUtils.isMatViewStateFileExists(configuration, path, tableToken.getDirName());
-                        path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
-                        if (isMatViewStateExists) {
-                            reader.of(path.$());
-                            MatViewRefreshState.readFrom(reader, state);
-                        } else {
-                            blockFileWriter.of(path.$());
-                            MatViewRefreshState.append(state, blockFileWriter);
-                        }
+                        // can be null if the graph has no-op implementation
+                        if (state != null) {
+                            final boolean isMatViewStateExists = TableUtils.isMatViewStateFileExists(configuration, path, tableToken.getDirName());
+                            path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
+                            if (isMatViewStateExists) {
+                                reader.of(path.$());
+                                MatViewRefreshState.readFrom(reader, state);
+                            } else {
+                                blockFileWriter.of(path.$());
+                                MatViewRefreshState.append(state, blockFileWriter);
+                            }
 
-                        if (!state.isInvalid()) {
-                            matViewGraph.enqueueIncrementalRefresh(tableToken);
+                            if (!state.isInvalid()) {
+                                matViewGraph.enqueueIncrementalRefresh(tableToken);
+                            }
```

Comment:
not in this PR, but log line below, lets change "error" to "errorMessage", there is literally one character difference between "error" and "errno"

Reply:
Changed to `msg` here in all similar places to keep things consistent with `QueryProgress` and other log calls:
5c9ffee4eb1dd2f32ae510faae082dcbd987f98a